### PR TITLE
Simplify packaging using Poetry and GitHub CI

### PR DESCRIPTION
**Pull request by jayrbolton**
Took out a lot of extra files and code in favor of using python-poetry, organizing the repo into looking more like a standard Python package, and using Github Actions for CI.

### CODE REVIEW

File: test/test_jsonrpcbase.py [hunk 1]
```diff
@@ -51,9 +51,9 @@ def test_multiple_args():
     """
     Test valid jsonrpc multiple argument calls.
     """
-    result = s.call_py('{"jsonrpc": "' +
-                       jsonrpcbase.DEFAULT_JSONRPC +
-                       '", "method": "subtract", "params": [42, 23], "id": "1"}')
+    result = s.call_py('{"jsonrpc": "'
```
Comment (jayrbolton) [inline test/test_jsonrpcbase.py:54]: All changes in this file are just flake8 autofix

File: jsonrpcbase/main.py [hunk 2]
```diff
@@ -448,8 +425,8 @@ def _validate_params_types(self, method, params):
 
             for key in params.keys():
                 if key not in self.method_data[method]['types'] or \
-                        not (isinstance(params[key], self.method_data[method]['types'][key]) or
-                             params[key] is None):
+                        not (isinstance(params[key], self.method_data[method]['types'][key])
+                             or params[key] is None):  # noqa
```
Comment (xpe) [inline jsonrpcbase/main.py]: what do you think about being more specific here? e.g. `# noqa: W503` (or W504)?

Comment (xpe) [reply jsonrpcbase/main.py]: ah, I see now that a [later commit](https://github.com/kbaseIncubator/jsonrpcbase/pull/1/commits/f95885d8c593a26f83caec360cc05791a96e1b58) removes the `# noqa` completely;

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
```
Comment (xpe) [inline CHANGELOG.md:7]: As discussed in Slack, how about having `## [Unreleased]` here?

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+## [0.3.0] - 2020-05-21
```
Comment (xpe) [inline CHANGELOG.md]: ... and moving these notes into 'unreleased' above?

**Comment by jayrbolton** [comment]
Moved changelog notes into the [Unreleased] section and removed the metadata feature note that isn't actually in this branch.

**Comment by jayrbolton** [comment]
@xpe for packaging, I've used setup.py and sdist, sdist + Pipenv, and later poetry. Poetry has been by far the most streamlined workflow for dependencies and publishing to pypi, so it's my top choice.

Pipenv feels quite similar to poetry, but poetry has a couple advantages:
* I like poetry's config file a lot
* `poetry build` and `poetry publish` is crazy easy and does not exist with pipenv.
* Poetry obviates the need to maintain a setup.py file

**Comment by xpe** [comment]
Minor/nit: I think "Simplify packaging using Poetry and GitHub CI." would be a more descriptive PR title.

**Comment by xpe** [comment]
`poetry check` reports `All set!` on my machine.

**Review by xpe — APPROVED** [review]
LGTM. Poetry looks good and the Changelog is a welcome addition.

**Pull request merged**